src/AgreementResolver.php

```diff
@@ -9,56 +9,94 @@
 
 class AgreementResolver {
   protected $storage;
+  protected $config;
```

Comment:
Mainly the change in this file is that cache is no longer set by route or role, but whole route and role collection is put into a single key.
This had to be done, so that it's easy to recreate the entire cache if the it gets erased by some point. So now the config cache gets created on request and not on drupal rebuild since the redis cache can be pretty volatile, meaning that we should *not* expect cache have alwasys certain values available, I mean it's just a "cache".  ;)